### ISSUE TITLE
Mac: Quick Look in Upload File Dialogue Cannot Be Closed

### ISSUE THREAD

**Issue by jasonblais** [mattermost/desktop#498]
I confirm (by marking "x" in the [ ] below): 

- [X] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [X] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [X] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**
Mac: File Preview in Upload File Dialogue Cannot Be Closed

When using the file upload dialog, opting to preview a file causes the preview window to appear. Once loaded, it cannot be closed cancelling upload entirely. Subsequently, Mattermost tends to quit unexpectedly.

**Steps to reproduce**

Mac OS 10.11.6; Mattermost Desktop App v3.5 and v3.6

Open the file upload dialogue, select a file, right-click Quick Look.

**Expected behavior**

Expected behavior is that clicking the close button icon, closes the Quick Look dialogue.

**Observed behavior**

Unable to close, except by cancelling upload. Subsequently, Mattermost tends to crash.

**Comment by yuya-oc** [comment]
The point is that the dialog is attached to the main window on macOS. It prevents our event handlers, so the webview couldn't get focus correctly.

To solve this, an event which is not supported by Electron is necessary. I confirmed the problem was solved by using modified Electron. So I'll prepare a PR for this.

But probably it doesn't meet the next milestone even if the PR is merged (it would be still beta when we release the next one).
